src/desktop/DesktopUtils.js

```diff
@@ -67,6 +71,54 @@ export default class DesktopUtils {
 		}
 	}
 
+	/**
+	 * take array of file names and add numbered suffixes to the basename of
+	 * duplicates. Use to make a legal set of names for files written to disk
+	 * at the same time.
+	 *
+	 * treats file names that already have numbered suffixes as non-numbered.
+	 * assumes the file system is case insensitive (a.txt would overwrite A.TXT)
+	 *
+	 * @param files file names.
+	 * @returns map from old names to array of new names. use map[oldname].shift() to replace oldname with newname.
+	 */
+	static legalizeFilenames(files: Array<string>): {[string]: Array<string>} {
+		const suffix = (name, suf) => {
+			const ext = path.extname(name)
+			const basename = path.basename(name, ext)
+			return `${basename}-${suf}${ext}`
```

Comment:
Is dot included in extension already? If you, we should probably leave a comment.